Source/AliveLib/Scrab.cpp

```diff
@@ -627,8 +628,483 @@ __int16 Scrab::AI_Patrol_0_4AA630()
+                                return 1;
+                            }
+                            int v20;
+                            if (FP_GetExponent(field_B8_xpos) - SnapToXGrid_449930(field_CC_sprite_scale, FP_GetExponent(field_B8_xpos)) >= 0)
+                            {
+                                v20 = FP_GetExponent(field_B8_xpos) - SnapToXGrid_449930(field_CC_sprite_scale, FP_GetExponent(field_B8_xpos));
+                            }
+                            else
+                            {
+                                v20 = SnapToXGrid_449930(field_CC_sprite_scale, FP_GetExponent(field_B8_xpos)) - FP_GetExponent(field_B8_xpos);
+                            }
+                            FP v23;
+                            if (v20 < 6
+                                && BaseAliveGameObject::Check_IsOnEndOfLine_408E90(field_C4_velx < FP_FromInteger(0), 1)
+                                && ((v6->field_BC_ypos - field_BC_ypos < FP_FromInteger(5))
+                                || ((field_C4_velx >= FP_FromInteger(0)) ? (v23 = ScaleToGridSize_4498B0(field_CC_sprite_scale)) : (v23 = -ScaleToGridSize_4498B0(field_CC_sprite_scale)),
+                                    sPath_dword_BB47C0->TLV_Get_At_4DB4B0(
+                                    FP_GetExponent(field_B8_xpos + v23),
+                                    FP_GetExponent(field_BC_ypos + FP_FromInteger(10)),
+                                    FP_GetExponent(field_B8_xpos + v23),
+                                    FP_GetExponent(field_BC_ypos + FP_FromInteger(10)),
+                                    TlvTypes::ElectricWall_38)))
+                                && !BaseAliveGameObject::Check_IsOnEndOfLine_408E90(field_C4_velx < FP_FromInteger(0), 3))
+                            {
+                                Scrab::ToJump_4A75E0();
+                                field_108_next_motion = -1;
+                                result = 7;
+                            }
+                            else if (vIsFacingMe_4254A0(v6))
+                            {
+                                if (!vIsObjNearby_4253B0(ScaleToGridSize_4498B0(field_CC_sprite_scale) / FP_FromInteger(7), v6)
+                                    || !v6->field_114_flags.Get(Flags_114::e114_Bit8_bInvisible) 
+                                    || v6->field_4_typeId != Types::eScrab_112)
+                                {
+                                    if (vIsObjNearby_4253B0(ScaleToGridSize_4498B0(field_CC_sprite_scale) * FP_FromInteger(3), v6)
+                                        && field_106_current_motion == eScrabMotions::M_Run_2_4A89C0
+                                        && vOnSameYLevel_425520(v6))
+                                    {
+                                        if (BaseAliveGameObject::WallHit_408750(field_CC_sprite_scale * FP_FromInteger(45), v6->field_B8_xpos - field_B8_xpos))

```

Comment:
BaseAliveGameObject:: is redundant ClassName:: is always redundant if the function isn't `static`